skimage/filters/_sparse.py

```diff
@@ -32,6 +31,89 @@ def _to_np_mode(mode):
     return mode
 
 
+def _validate_window_size(axis_sizes):
+    """Ensure all sizes in ``axis_sizes`` are odd.
+
+    Parameters
+    ----------
+    axis_sizes : iterable of int
+
+    Raises
+    ------
+    ValueError
+        If any given axis size is even.
+    """
+    for axis_size in axis_sizes:
+        if axis_size % 2 == 0:
+            msg = ('Window size must not be even on any dimension. '
+                   'Got {}'.format(axis_sizes))
+            raise ValueError(msg)
+
+
+def _to_np_mode(mode):
+    """Convert padding modes from `ndi.correlate` to `np.pad`."""
+    mode_translation_dict = dict(nearest='edge', reflect='symmetric',
+                                 mirror='reflect')
+    if mode in mode_translation_dict:
+        mode = mode_translation_dict[mode]
+    return mode
```

Comment:
```suggestion

```

`_validate_window_size` and `_to_np_mode` are already defined above :wink:.

BTW, `_to_np_mode` is already defined in `skimage.filters._sparse` and the opposite function `_to_ndimage_mode` is defined in `skimage.transform._geometric` and used in `skimage.tranform._warp`. These two functions can be added to `skimage._shared` module.